tls/s2n_config.c

```diff
@@ -808,6 +812,30 @@ int s2n_config_set_extension_data(struct s2n_config *config, s2n_tls_extension_t
     return 0;
 }
 
+int s2n_config_set_custom_critical_oids(struct s2n_config *config, const char *const *custom_critical_oids, uint32_t custom_oid_count)
+{
+    POSIX_ENSURE_REF(config);
+
+#ifndef S2N_LIBCRYPTO_SUPPORTS_CUSTOM_OID
+    POSIX_BAIL(S2N_ERR_INTERNAL_LIBCRYPTO_ERROR);
```

Comment:
If the user is trying to use this feature but the libcrypto doesn't support it, this should probably be considered a usage error. `S2N_ERR_API_UNSUPPORTED_BY_LIBCRYPTO` might be more appropriate.